WeakAuras/RegionTypes/DynamicGroup.lua

```diff
@@ -277,17 +326,27 @@ local growers = {
     local limit = data.useLimit and data.limit or math.huge
     local startX, startY = 0, 0
     local coeff = staggerCoefficient(data.align, data.stagger)
+    local anchorPerUnitFunc = data.useAnchorPerUnit and createAnchorPerUnitFunc(data)
     return function(newPositions, activeRegions)
-      local numVisible = min(limit, #activeRegions)
-      local x, y = startX, startY + (numVisible - 1) * stagger * coeff
-      local i = 1
-      while i <= numVisible do
-        local pos = {x, y}
-        local regionData = activeRegions[i]
-        newPositions[i] = pos
-        x = x - (regionData.data.width or regionData.region.width) - space
-        y = y - stagger
-        i = i + 1
+      local frames = {}
+      if anchorPerUnitFunc then
+        anchorPerUnitFunc(frames, activeRegions)
+      else
+        frames[""] = activeRegions
+      end
+      for frame, regionDatas in pairs(frames) do
+        local numVisible = min(limit, #regionDatas)
+        local x, y = startX, startY + (numVisible - 1) * stagger * coeff
+        newPositions[frame] = {}
+        for i, regionData in ipairs(regionDatas) do
+          if i <= numVisible then
+            newPositions[frame][regionData] = { x, y, true }
+            x = x - (regionData.data.width or regionData.region.width) - space
+            y = y - stagger
+          else
+            newPositions[frame][regionData] = { [3] = false }
```

Comment:
This does too much work. If you've exceeded the limit, then don't add position data for that regionData object. The positioning system is smart enough to detect that you just want to hide it.

Setting the show value to false is useful only for making animated dynamic groups behave slightly more consistently.